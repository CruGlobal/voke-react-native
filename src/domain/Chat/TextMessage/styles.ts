import { StyleSheet } from 'react-native';
import theme from 'utils/theme';

const sharedStyles = {
  avatar: {
    position: 'absolute' as const,
    bottom: 0,
    height: 25,
    width: 25,
    borderRadius: 50,
  },
  messageBubble: {
    width: '100%',
    borderRadius: 8,
    overflow: 'hidden' as const, // Needed to hide overflow blur eff.
  },
};

const styles = StyleSheet.create({
  container: {
    // align 'between'
    paddingHorizontal: theme.spacing.l,
    paddingBottom: theme.spacing.m,
    zIndex: 1, // Needed to make context menu covering all other elements.
  },
  content: {
    paddingHorizontal: theme.spacing.l,
  },
  messageBubbleOthers: {
    ...sharedStyles.messageBubble,
    backgroundColor: theme.colors.secondary,
  },
  messageBubbleMine: {
    ...sharedStyles.messageBubble,
    backgroundColor: theme.colors.white,
  },
  author: {
    color: theme.colors.white,
    paddingTop: theme.spacing.m,
    paddingHorizontal: theme.spacing.l,
    marginBottom: -theme.spacing.m,
    fontFamily: theme.fonts.semiBold,
    opacity: 0.75,
  },
  sharedContent: {
    marginTop: theme.spacing.l,
    padding: theme.spacing.l,
  },
  sharedText: {
    fontSize: theme.fontSizes.m,
  },
  messageText: {
    paddingHorizontal: theme.spacing.l,
    paddingVertical: theme.spacing.l,
    fontSize: theme.fontSizes.l,
  },
  myAvatar: {
    ...sharedStyles.avatar,
    right: -30,
  },
  userAvatar: {
    ...sharedStyles.avatar,
    left: -30,
    backgroundColor: theme.colors.secondaryAlt,
  },
  messageMeta: {
    paddingTop: theme.spacing.xs,
    flexDirection: 'row',
  },
  date: {
    textAlign: 'right',
    fontSize: theme.fontSizes.xs,
    color: theme.colors.white,
  },
});

export default styles;
