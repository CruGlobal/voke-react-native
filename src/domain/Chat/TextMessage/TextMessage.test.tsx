import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import { Host } from 'react-native-portalize';
import '@testing-library/jest-native';
import { messengerMock, mockMessage, mockUser } from 'mocks/vokeDataMocks';
import renderWithContext from 'utils/testUtils';

import TextMessage from './index';

const initialState = {
  auth: { user: mockUser },
  data: {},
};

it('renders correctly', () => {
  const { queryByText, toJSON } = renderWithContext(
    <TextMessage
      user={mockUser}
      message={mockMessage}
      messenger={messengerMock}
      isBlured={false}
      isSharedAnswer={false}
      canReport={true}
      onReport={(): void => {
        // void.
      }}
      onCopy={(): void => {
        // void.
      }}
      onReaction={(): void => {
        // void.
      }}
      setContextActive={(): void => {
        // void.
      }}
    />,
    { initialState },
  );

  expect(toJSON()).toMatchSnapshot();
  expect(queryByText(/John Doe/i)).toBeTruthy();
  expect(queryByText(/Test message content/i)).toBeTruthy();
  expect(queryByText(/Jan 25/i)).toBeTruthy();
  expect(queryByText(/Report/i)).not.toBeTruthy();
  expect(queryByText(/Copy/i)).not.toBeTruthy();
});

it('context mode renders correctly', () => {
  const { queryByText, queryByTestId, getByTestId, toJSON } = renderWithContext(
    <Host>
      <TextMessage
        user={mockUser}
        message={mockMessage}
        messenger={messengerMock}
        isBlured={false}
        isSharedAnswer={false}
        canReport={true}
        contextActive={null}
        onReport={(): void => {
          // void.
        }}
        onCopy={(): void => {
          // void.
        }}
        onReaction={(): void => {
          // void.
        }}
        setContextActive={(): void => {
          // void.
        }}
      />
    </Host>,
    { initialState },
  );
  // Activate message context mode.
  fireEvent(getByTestId('messagePressArea'), 'onLongPress');
  // Do all the checks.
  expect(toJSON()).toMatchSnapshot();
  expect(queryByText(/John Doe/i)).toBeTruthy();
  expect(queryByText(/Test message content/i)).toBeTruthy();
  expect(queryByText(/Jan 25/i)).toBeTruthy();
  expect(queryByTestId('reactionsPopup')).toBeTruthy();
  expect(queryByText(/Report/i)).toBeTruthy();
  expect(queryByText(/Copy/i)).toBeTruthy();
});

it('reactions calculated correctly', () => {
  const { queryByText } = renderWithContext(
    <TextMessage
      user={mockUser}
      message={mockMessage}
      messenger={messengerMock}
      isBlured={false}
      isSharedAnswer={false}
      contextActive={null}
      canReport={true}
      onReport={(): void => {
        // void.
      }}
      onCopy={(): void => {
        // void.
      }}
      onReaction={(): void => {
        // void.
      }}
      setContextActive={(): void => {
        // void.
      }}
    />,
    { initialState },
  );

  expect(queryByText(/🤗 3/i)).toBeTruthy();
  expect(queryByText(/😄 1/i)).toBeTruthy();
  expect(queryByText(/♥️ 3/i)).toBeTruthy();
  expect(queryByText(/😂 1/i)).toBeTruthy();
});

it('emit events correctly', () => {
  let contextActiveState = null;
  const onSetContextActive = () => {
    contextActiveState = contextActiveState ? null : mockMessage.id;
  };
  const onReport = jest.fn();
  const onCopy = jest.fn();
  const onReaction = jest.fn(() => onSetContextActive());
  const { queryByTestId, getByTestId, rerender } = renderWithContext(
    <Host>
      <TextMessage
        user={mockUser}
        message={mockMessage}
        messenger={messengerMock}
        isBlured={false}
        isSharedAnswer={false}
        canReport={true}
        onReport={onReport}
        onCopy={onCopy}
        onReaction={(newReaction): void => {
          onReaction(newReaction);
        }}
        contextActive={contextActiveState}
        setContextActive={(): void => {
          onSetContextActive();
        }}
      />
    </Host>,
    { initialState },
  );
  // Activate context mode.
  fireEvent(getByTestId('messagePressArea'), 'onLongPress');

  rerender(
    <Host>
      <TextMessage
        user={mockUser}
        message={mockMessage}
        messenger={messengerMock}
        isBlured={false}
        isSharedAnswer={false}
        canReport={true}
        onReport={onReport}
        onCopy={onCopy}
        onReaction={(newReaction): void => {
          onReaction(newReaction);
        }}
        contextActive={contextActiveState}
        setContextActive={(): void => {
          onSetContextActive();
        }}
      />
    </Host>,
  );
  expect(queryByTestId('reactionsPopup')).toBeTruthy();

  fireEvent(getByTestId('reaction-1'), 'onPressIn');
  expect(onReaction).toHaveBeenCalled();
  rerender(
    <Host>
      <TextMessage
        user={mockUser}
        message={mockMessage}
        messenger={messengerMock}
        isBlured={false}
        isSharedAnswer={false}
        canReport={true}
        onReport={onReport}
        onCopy={onCopy}
        onReaction={(newReaction): void => {
          onReaction(newReaction);
        }}
        contextActive={contextActiveState}
        setContextActive={(): void => {
          onSetContextActive();
        }}
      />
    </Host>,
  );
  expect(queryByTestId('reactionsPopup')).not.toBeTruthy(); // Context mode close.

  fireEvent.press(getByTestId('reactionPill-1'));
  expect(onReaction).toHaveBeenCalled();

  // Open context mode again.
  fireEvent(getByTestId('messagePressArea'), 'onLongPress');
  fireEvent.press(getByTestId('report'));
  expect(onReport).toHaveBeenCalled();
  expect(queryByTestId('report')).not.toBeTruthy(); // Context mode close.

  // Open context mode again.
  fireEvent(getByTestId('messagePressArea'), 'onLongPress');
  fireEvent.press(getByTestId('copy'));
  expect(onCopy).toHaveBeenCalled();
  expect(queryByTestId('copy')).not.toBeTruthy(); // Context mode close.
});

it('emit reaction pill events correctly', () => {
  const onReaction = jest.fn();
  let contextActiveState = null;
  const onSetContextActive = () => {
    contextActiveState = mockMessage.id;
  };
  const { getByText } = renderWithContext(
    <Host>
      <TextMessage
        user={mockUser}
        message={mockMessage}
        messenger={messengerMock}
        isBlured={false}
        isSharedAnswer={false}
        canReport={true}
        onReport={(): void => {
          // void
        }}
        onCopy={(): void => {
          // void
        }}
        onReaction={(newReaction): void => {
          onReaction(newReaction);
        }}
        contextActive={contextActiveState}
        setContextActive={(): void => {
          onSetContextActive();
        }}
      />
    </Host>,
    { initialState },
  );

  fireEvent.press(getByText(/♥️/i));
  expect(onReaction).toHaveBeenCalledWith('♥️');

  fireEvent.press(getByText(/😄/i));
  expect(onReaction).toHaveBeenCalledWith('😄');

  fireEvent.press(getByText(/😂/i));
  expect(onReaction).toHaveBeenCalledWith('😂');

  fireEvent.press(getByText(/🤗/i));
  expect(onReaction).toHaveBeenCalledWith('🤗');
});
