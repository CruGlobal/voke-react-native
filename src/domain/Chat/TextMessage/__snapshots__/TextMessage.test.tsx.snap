// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`context mode renders correctly 1`] = `
Array [
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "paddingBottom": 16,
            "paddingHorizontal": 22,
            "zIndex": 1,
          },
          Object {
            "zIndex": 1,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "paddingHorizontal": 22,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "rgba(0,0,0,.6)",
              "bottom": -1000,
              "left": -1000,
              "position": "absolute",
              "right": -1000,
              "top": -1000,
              "zIndex": 1,
            }
          }
        />
        <View
          forwardedRef={[Function]}
          style={
            Object {
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
              "width": "100%",
              "zIndex": 2,
            }
          }
        >
          <View>
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "backgroundColor": "#186078",
                  "borderRadius": 8,
                  "overflow": "hidden",
                  "width": "100%",
                }
              }
              testID="messagePressArea"
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "#fff",
                      "fontFamily": "TitilliumWeb-SemiBold",
                      "marginBottom": -16,
                      "opacity": 0.75,
                      "paddingHorizontal": 22,
                      "paddingTop": 16,
                    }
                  }
                >
                  John 
                  Doe 
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "fontSize": 17,
                        "paddingHorizontal": 22,
                        "paddingVertical": 22,
                      },
                      Object {
                        "color": "#fff",
                        "opacity": 1,
                      },
                    ]
                  }
                >
                  Test message content
                </Text>
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              />
            </View>
            <Image
              source={
                Object {
                  "uri": "https://assets.vokeapp.com/images/user/small/avatar.jpg",
                }
              }
              style={
                Object {
                  "backgroundColor": "#3295ad",
                  "borderRadius": 50,
                  "bottom": 0,
                  "height": 25,
                  "left": -30,
                  "position": "absolute",
                  "width": 25,
                }
              }
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "paddingTop": 4,
                },
                Object {
                  "justifyContent": "flex-start",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#1d1d26",
                  },
                  Object {
                    "fontFamily": "TitilliumWeb-Regular",
                  },
                  Object {
                    "color": "#fff",
                    "fontSize": 13,
                    "textAlign": "right",
                  },
                ]
              }
            >
              Jan 25 @ 4:44 PM
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "flex-start",
                    "alignSelf": "flex-start",
                    "flex": 1,
                    "flexDirection": "row",
                    "flexWrap": "wrap",
                  },
                  Object {
                    "justifyContent": "flex-end",
                  },
                ]
              }
            >
              <View
                forwardedRef={[Function]}
                style={Object {}}
              >
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "borderRadius": 12,
                      "marginBottom": 4,
                      "marginLeft": 8,
                      "paddingHorizontal": 4,
                      "paddingVertical": 4,
                    }
                  }
                  testID="reactionPill-0"
                >
                  <Text
                    style={
                      Object {
                        "color": "#186078",
                        "fontSize": 11,
                        "paddingRight": 4,
                      }
                    }
                  >
                    🤗
                     
                    3
                  </Text>
                </View>
              </View>
              <View
                forwardedRef={[Function]}
                style={Object {}}
              >
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "borderRadius": 12,
                      "marginBottom": 4,
                      "marginLeft": 8,
                      "paddingHorizontal": 4,
                      "paddingVertical": 4,
                    }
                  }
                  testID="reactionPill-1"
                >
                  <Text
                    style={
                      Object {
                        "color": "#186078",
                        "fontSize": 11,
                        "paddingRight": 4,
                      }
                    }
                  >
                    😄
                     
                    1
                  </Text>
                </View>
              </View>
              <View
                forwardedRef={[Function]}
                style={Object {}}
              >
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "borderRadius": 12,
                      "marginBottom": 4,
                      "marginLeft": 8,
                      "paddingHorizontal": 4,
                      "paddingVertical": 4,
                    }
                  }
                  testID="reactionPill-2"
                >
                  <Text
                    style={
                      Object {
                        "color": "#186078",
                        "fontSize": 11,
                        "paddingRight": 4,
                      }
                    }
                  >
                    ♥️
                     
                    3
                  </Text>
                </View>
              </View>
              <View
                forwardedRef={[Function]}
                style={Object {}}
              >
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "borderRadius": 12,
                      "marginBottom": 4,
                      "marginLeft": 8,
                      "paddingHorizontal": 4,
                      "paddingVertical": 4,
                    }
                  }
                  testID="reactionPill-3"
                >
                  <Text
                    style={
                      Object {
                        "color": "#186078",
                        "fontSize": 11,
                        "paddingRight": 4,
                      }
                    }
                  >
                    😂
                     
                    1
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            forwardedRef={[Function]}
            style={
              Object {
                "opacity": 0,
                "position": "absolute",
                "top": -76,
                "transform": Array [
                  Object {
                    "translateY": 100,
                  },
                ],
                "width": "100%",
              }
            }
            testID="reactionsPopup"
          >
            <View
              style={
                Object {
                  "backgroundColor": "#fefefe",
                  "borderRadius": 12,
                  "flexDirection": "row",
                  "height": 60,
                  "justifyContent": "space-evenly",
                  "paddingHorizontal": 8,
                }
              }
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "stretch",
                    "alignSelf": "stretch",
                    "flex": 1,
                    "height": 60,
                    "justifyContent": "center",
                    "textAlign": "center",
                  }
                }
                testID="reaction-0"
              >
                <Text
                  style={
                    Object {
                      "fontSize": 38,
                      "lineHeight": 60,
                      "textAlign": "center",
                    }
                  }
                >
                  ♥️
                </Text>
              </View>
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "stretch",
                    "alignSelf": "stretch",
                    "flex": 1,
                    "height": 60,
                    "justifyContent": "center",
                    "textAlign": "center",
                  }
                }
                testID="reaction-1"
              >
                <Text
                  style={
                    Object {
                      "fontSize": 38,
                      "lineHeight": 60,
                      "textAlign": "center",
                    }
                  }
                >
                  😄
                </Text>
              </View>
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "stretch",
                    "alignSelf": "stretch",
                    "flex": 1,
                    "height": 60,
                    "justifyContent": "center",
                    "textAlign": "center",
                  }
                }
                testID="reaction-2"
              >
                <Text
                  style={
                    Object {
                      "fontSize": 38,
                      "lineHeight": 60,
                      "textAlign": "center",
                    }
                  }
                >
                  😂
                </Text>
              </View>
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "stretch",
                    "alignSelf": "stretch",
                    "flex": 1,
                    "height": 60,
                    "justifyContent": "center",
                    "textAlign": "center",
                  }
                }
                testID="reaction-3"
              >
                <Text
                  style={
                    Object {
                      "fontSize": 38,
                      "lineHeight": 60,
                      "textAlign": "center",
                    }
                  }
                >
                  🤗
                </Text>
              </View>
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "stretch",
                    "alignSelf": "stretch",
                    "flex": 1,
                    "height": 60,
                    "justifyContent": "center",
                    "textAlign": "center",
                  }
                }
                testID="reaction-4"
              >
                <Text
                  style={
                    Object {
                      "fontSize": 38,
                      "lineHeight": 60,
                      "textAlign": "center",
                    }
                  }
                >
                  😓
                </Text>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "height": 40,
                    "transform": Array [
                      Object {
                        "rotate": "0deg",
                      },
                    ],
                    "width": 40,
                  },
                  Object {
                    "bottom": -10,
                    "left": 14,
                    "position": "absolute",
                    "zIndex": -1,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderColor": "transparent",
                      "borderLeftColor": "#eee",
                      "borderLeftWidth": 20,
                      "borderTopWidth": 20,
                      "borderWidth": 0,
                    },
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "transform": Array [
                        Object {
                          "scale": -1,
                        },
                      ],
                    },
                  ]
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderColor": "transparent",
                      "borderLeftColor": "#eee",
                      "borderLeftWidth": 20,
                      "borderTopWidth": 20,
                      "borderWidth": 0,
                    },
                    Object {
                      "bottom": 0,
                      "position": "absolute",
                      "right": 0,
                      "transform": Array [
                        Object {
                          "scaleY": -1,
                        },
                      ],
                    },
                  ]
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>,
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      forwardedRef={[Function]}
      style={
        Object {
          "bottom": 0,
          "height": 90,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "transform": Array [
            Object {
              "translateY": 100,
            },
          ],
          "zIndex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#44c8e8",
            "flexDirection": "row",
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <RNCSafeAreaView
          edges={
            Array [
              "bottom",
              "right",
              "left",
            ]
          }
          style={
            Object {
              "flexDirection": "row",
              "width": "100%",
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
            testID="report"
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#1d1d26",
                  },
                  Object {
                    "fontFamily": "TitilliumWeb-Regular",
                  },
                  Array [
                    Object {
                      "opacity": 1,
                    },
                    Object {
                      "color": "#fff",
                      "fontSize": 17,
                    },
                  ],
                ]
              }
            >
              report
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
            testID="copy"
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#1d1d26",
                  },
                  Object {
                    "fontFamily": "TitilliumWeb-Regular",
                  },
                  Object {
                    "color": "#fff",
                    "fontSize": 17,
                  },
                ]
              }
            >
              copy
            </Text>
          </View>
        </RNCSafeAreaView>
      </View>
    </View>
  </View>,
]
`;

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "paddingBottom": 16,
        "paddingHorizontal": 22,
        "zIndex": 1,
      },
      Object {
        "zIndex": -1,
      },
    ]
  }
>
  <View
    style={
      Object {
        "paddingHorizontal": 22,
      }
    }
  >
    <View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#186078",
            "borderRadius": 8,
            "overflow": "hidden",
            "width": "100%",
          }
        }
        testID="messagePressArea"
      >
        <View>
          <Text
            style={
              Object {
                "color": "#fff",
                "fontFamily": "TitilliumWeb-SemiBold",
                "marginBottom": -16,
                "opacity": 0.75,
                "paddingHorizontal": 22,
                "paddingTop": 16,
              }
            }
          >
            John 
            Doe 
          </Text>
          <Text
            style={
              Array [
                Object {
                  "fontSize": 17,
                  "paddingHorizontal": 22,
                  "paddingVertical": 22,
                },
                Object {
                  "color": "#fff",
                  "opacity": 1,
                },
              ]
            }
          >
            Test message content
          </Text>
        </View>
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        />
      </View>
      <Image
        source={
          Object {
            "uri": "https://assets.vokeapp.com/images/user/small/avatar.jpg",
          }
        }
        style={
          Object {
            "backgroundColor": "#3295ad",
            "borderRadius": 50,
            "bottom": 0,
            "height": 25,
            "left": -30,
            "position": "absolute",
            "width": 25,
          }
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "paddingTop": 4,
          },
          Object {
            "justifyContent": "flex-start",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#1d1d26",
            },
            Object {
              "fontFamily": "TitilliumWeb-Regular",
            },
            Object {
              "color": "#fff",
              "fontSize": 13,
              "textAlign": "right",
            },
          ]
        }
      >
        Jan 25 @ 4:44 PM
      </Text>
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "alignSelf": "flex-start",
              "flex": 1,
              "flexDirection": "row",
              "flexWrap": "wrap",
            },
            Object {
              "justifyContent": "flex-end",
            },
          ]
        }
      >
        <View
          forwardedRef={[Function]}
          style={Object {}}
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 12,
                "marginBottom": 4,
                "marginLeft": 8,
                "paddingHorizontal": 4,
                "paddingVertical": 4,
              }
            }
            testID="reactionPill-0"
          >
            <Text
              style={
                Object {
                  "color": "#186078",
                  "fontSize": 11,
                  "paddingRight": 4,
                }
              }
            >
              🤗
               
              3
            </Text>
          </View>
        </View>
        <View
          forwardedRef={[Function]}
          style={Object {}}
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 12,
                "marginBottom": 4,
                "marginLeft": 8,
                "paddingHorizontal": 4,
                "paddingVertical": 4,
              }
            }
            testID="reactionPill-1"
          >
            <Text
              style={
                Object {
                  "color": "#186078",
                  "fontSize": 11,
                  "paddingRight": 4,
                }
              }
            >
              😄
               
              1
            </Text>
          </View>
        </View>
        <View
          forwardedRef={[Function]}
          style={Object {}}
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 12,
                "marginBottom": 4,
                "marginLeft": 8,
                "paddingHorizontal": 4,
                "paddingVertical": 4,
              }
            }
            testID="reactionPill-2"
          >
            <Text
              style={
                Object {
                  "color": "#186078",
                  "fontSize": 11,
                  "paddingRight": 4,
                }
              }
            >
              ♥️
               
              3
            </Text>
          </View>
        </View>
        <View
          forwardedRef={[Function]}
          style={Object {}}
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 12,
                "marginBottom": 4,
                "marginLeft": 8,
                "paddingHorizontal": 4,
                "paddingVertical": 4,
              }
            }
            testID="reactionPill-3"
          >
            <Text
              style={
                Object {
                  "color": "#186078",
                  "fontSize": 11,
                  "paddingRight": 4,
                }
              }
            >
              😂
               
              1
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
